File: app/src/main/java/de/karzek/diettracker/presentation/main/settings/dialog/editAllergen/adapter/viewHolder/AllergenViewHolder.java
package de.karzek.diettracker.presentation.main.settings.dialog.editAllergen.adapter.viewHolder;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.CheckBox;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import de.karzek.diettracker.R;
import de.karzek.diettracker.presentation.model.AllergenDisplayModel;

/**
 * Created by MarjanaKarzek on 30.05.2018.
 *
 * @author Marjana Karzek
 * @version 1.0
 * @date 30.05.2018
 */
public class AllergenViewHolder extends RecyclerView.ViewHolder {

    @BindView(R.id.checkbox) CheckBox allergen;

    private final OnItemCheckedChangeListener onItemCheckedChangeListener;

    public AllergenViewHolder(ViewGroup viewGroup,
                              OnItemCheckedChangeListener onItemCheckedChangeListener) {
        super(LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.item_dialog_checkbox, viewGroup, false));
        ButterKnife.bind(this, itemView);
        this.onItemCheckedChangeListener = onItemCheckedChangeListener;
    }

    public void bind(AllergenDisplayModel displayModel, boolean status) {
        itemView.setTag(displayModel.getId());

        allergen.setText(displayModel.getName());
        allergen.setChecked(status);
    }

    @OnClick (R.id.checkbox) public void onCheckedChanged(){
        onItemCheckedChangeListener.onItemCheckChanged((int) itemView.getTag(), allergen.isChecked());
    }

    public interface OnItemCheckedChangeListener {
        void onItemCheckChanged(int id, boolean checked);
    }

}

File: app/src/main/java/de/karzek/diettracker/presentation/main/settings/dialog/editAllergen/adapter/viewHolder/AllergenViewHolder.kt
package de.karzek.diettracker.presentation.main.settings.dialog.editAllergen.adapter.viewHolder

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import de.karzek.diettracker.presentation.model.AllergenDisplayModel
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.item_dialog_checkbox.*

/**
 * Created by MarjanaKarzek on 30.05.2018.
 *
 * @author Marjana Karzek
 * @version 1.0
 * @date 30.05.2018
 */

class AllergenViewHolder(override val containerView: View,
                         private val onItemCheckedChangeListener: OnItemCheckedChangeListener) : RecyclerView.ViewHolder(containerView), LayoutContainer {

    fun bind(displayModel: AllergenDisplayModel, status: Boolean) {
        itemView.tag = displayModel.id

        checkbox.text = displayModel.name
        checkbox.isChecked = status
        checkbox.setOnClickListener { onItemCheckedChangeListener.onItemCheckChanged(itemView.tag as Int, checkbox.isChecked) }
    }

    interface OnItemCheckedChangeListener {
        fun onItemCheckChanged(id: Int, checked: Boolean)
    }

}
